Nodejs/Product/TestAdapter/TestExecutorWorker.cs

```diff
@@ -91,6 +91,30 @@ public void RunTests(IEnumerable<TestCase> tests)
             ValidateArg.NotNull(tests, nameof(tests));
             this.cancelRequested.Reset();
 
+            var testFramework = tests.First().GetPropertyValue(JavaScriptTestCaseProperties.TestFramework, defaultValue: TestFrameworkDirectories.ExportRunnerFrameworkName);
+            if (testFramework == "Angular")
+            {
+                RunAllTests(tests);
+            }
+            else
```

Comment:
It is that in Angular, is better to run all of the tests in a given "project" (as defined in `angular.json` and `karma.conf.js`), otherwise we'll initialize karma, browsers, cache, etc, multiple times. I have added a small comment on the line.

Technically other languages don't need this processing. I'm not entirely sure why is implemented like this. I believe some time ago there used to be a push for parallelizing this process, but I prefer to let the test frameworks do their stuff (like in the case angular implementation).

Hope this answers your questions.